### ISSUE TITLE
Use the sync Span field

### ISSUE THREAD

**Issue by eyalkoren** [elastic/apm-agent-java#1239]
We are currently not utilizing the Span `sync` flag, which enables labelling spans as being executed synchronously or asynchronously.
A label will be shown for either `sync == true` or `sync == false`, so the scope of this issue is only about setting `sync == false` wherever applicable. 
In the future, we may want to also denote blocking spans within asynchronous frameworks.

**Comment by SylvainJuge** [comment]
Does it means that `sync` span value reflects whether or not it's blocking or not or is it something else ?

From my understanding:
- `sync=true` means that span is executed in a blocking way, calling code actively waits for it's termination, thus in the same thread.
- `sync=false` means that span is executed asynchronously from the code that triggered it, thus in another thread.

**Comment by SylvainJuge** [comment]
After checking with the APM UI team, there are 3 cases for the `span.sync` field:
- if not set (or `null`), then we assume it's using the "platform default" which is platform/agent dependent, in this case not badge is shown in the UI
- if set to `true`, then a `blocking` badge is shown in the UI
- if set to `false`, then an `async` badge is shown in the UI.

So in practice it means the values effectively used are platform-specific, but having an explicit value `true` or `false` means a badge is shown, which could definitely clutter the UI.
- in Java, it's either empty / `null` or `false` as the most common in java is blocking, we only show badges when it's async.
- with RUM, it is either empty / `null` or `true`, in this case we only show badges when it is blocking.

I have opened https://github.com/elastic/apm/pull/802 as an attempt to clarify this.

If we stick to this definition, then the only thing we have to do is to ensure that we set `sync=false` whenever it's applicable, by reviewing and modifying all asynchronous frameworks instrumentation that should create such spans (currently only a few AWS ones do that).
